### PR TITLE
filter by response & three other small features

### PR DESCRIPTION
**Pull request by ShanniLi**
r @Raynos @kriskowal 

Four changes:
1. add response status
Before: session=0 127.0.0.1:56901 --> 127.0.0.1:4040 frame=1 type=0x04
After: session=0 127.0.0.1:56901 --> 127.0.0.1:4040 frame=1 type=0x04 NotOk

2. add the message section if it is available int the frame, e.g.,
session=0 127.0.0.1:56775 <-- 127.0.0.1:4040 frame=2 type=0xff
ERROR[BadRequest] id=0x0002 (2)
tracing: spanid=0000000000000000 parentid=0000000000000000 traceid=0000000000000000 flags=0x00
message
    00: 2121 2121 216e 6f74 204f 4b21 2121 21    !!!!!not OK!!!!

3. add the error type for error responses. The example is shown up, i.e., ERROR[BadRequest]

4. Add a new filter: filter by response type: Ok, NotOk, and Error.

### CODE REVIEW

File: bin/tcap.js [hunk 1]
```diff
@@ -54,6 +54,9 @@ function main(argv) {
         .option('-1 --arg1 <arg1-method>',
             'arg1 method or methods to show ' +
             '(default: all arg1 methods shown)', collect, [])
+        .option('-r --response <response>',
+            'responses to show: O[K], N[otOk], E[rror] ' +
+            '(default: all shown)', collect, [])
```
Comment (kriskowal) [inline bin/tcap.js]: `-r` for "response" seems too generic. Perhaps we can break this into --ok, --not-ok, and --error flags.

File: tchannel-session-tracker.js [hunk 2]
```diff
@@ -154,6 +161,29 @@ function handleFrame(frame) {
...
+                    return;
+                }
+            } else if (frame.body.type === FrameType.Error) {
+                if (!self.responseStatuses[FrameType.Error]) {
+                    return;
+                }
+            } else {
+                return;
+            }
+        } else {
```
Comment (kriskowal) [inline tchannel-session-tracker.js]: Would be interesting, down the line, to suspend the request frames until the nature of the response is known.

File: tchannel-tracker.js [hunk 3]
```diff
@@ -57,6 +60,32 @@ function TChannelTracker(opts) {
...
+                    break;
+                default:
+                    console.log(ansi.red(
+                    ansi.bold('Warning: wrong response status ' +
+                              'in command options: \'-r ' +
+                              name + '\'')));
+                    break;
+            }
+        });
+    }
```
Comment (kriskowal) [inline tchannel-tracker.js]: Let’s not leak user-interface level concerns into the implementation. In bin/tcap.js, normalize all of these cases into an option or options.

File: tchannel-types.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+module.exports.ResponseType = {
+	'Ok': 0x00,
+	'NotOk': 0x01
+};
+
+var responseNameByType = [];
+responseNameByType[0x00] = 'Ok';
+responseNameByType[0x01] = 'NotOk';
+module.exports.ResponseNameByType = responseNameByType;
```
Comment (kriskowal) [inline tchannel-types.js]: All of this information can be imported from the module in tchannel with `require`.

**Comment by kriskowal** [comment]
The only feedback above that I would insist change before landing is to use `require` to import constants from `tchannel`.

File: tchannel-session-tracker.js (hunk 2 above)
Comment (ShanniLi) [inline tchannel-session-tracker.js]: I will make it a TODO item.

File: tchannel-tracker.js (hunk 3 above)
Comment (ShanniLi) [inline tchannel-tracker.js]: fixed.

File: tchannel-types.js (hunk 4 above)
Comment (ShanniLi) [inline tchannel-types.js]: fixed except some constant table not defined in tchannel.

File: bin/tcap.js (hunk 1 above)
Comment (ShanniLi) [inline bin/tcap.js]: I will keep it for now and refine it later.

**Pull request merged**